simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/parser/snia/systor/SystorTraceLatencyReader.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Omri Himelbrand. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.benmanes.caffeine.cache.simulator.parser.snia.systor;
+
+import com.github.benmanes.caffeine.cache.simulator.parser.TextTraceReader;
+import com.github.benmanes.caffeine.cache.simulator.policy.AccessEvent;
+import com.github.benmanes.caffeine.cache.simulator.policy.Policy.Characteristic;
+import com.google.common.collect.ImmutableSet;
+
+import java.math.BigInteger;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * A reader for the SNIA SYSTOR '17 trace files provided by
+ * <a href="http://iotta.snia.org/traces/4964">SNIA</a>.
+ * This reader uses the response time as miss penalty (converted to milliseconds).
+ *
+ * @author himelbrand@gmail.com (Omri Himelbrand)
+ */
+public final class SystorTraceLatencyReader extends TextTraceReader {
+
+  public SystorTraceLatencyReader(String filePath) {
+    super(filePath);
+  }
+
+  @Override
+  public Set<Characteristic> characteristics() {
+    return ImmutableSet.of();
+  }
+
+  @Override
+  public Stream<AccessEvent> events() {
+
+    return lines()
+            .map(line -> line.split(","))
+            .filter(array -> array[2].equals("R") && array[1].length() > 0)
+            .map(array -> {
+              long key = new BigInteger(array[4]).longValue();
```

Comment:
Just fyi that this truncates the number. For this workload I think it should be equivalent to being hashed, but I don't know if it causes skew compared to running it through a function like murmur3. You might want to try hashing to double check that that the hit rates are similar with either scheme.

```java
jshell> var b = BigInteger.valueOf(Long.MAX_VALUE).shiftLeft(2) // 4 x max
b ==> 36893488147419103228
jshell> var c = b.longValue()
c ==> -4
```

Reply:
You're righ about the possible truncation, I'm running a quick (as possible) analisys of the offsets field, checking the number of bits required to represent the largest offset, if it is less or equal than 64, then we are good with this implemantation.
If not, it might be better to use a hash.
I'll let you know as soon as Ihave the results of the analisys.

Reply:
A simple approach is to use toByteArray and guava’s hasher. If it’s similar than no reason to change.

I suppose since this is not split into blocks (e.g. like ARC’s) the chance of a hit is much lower. A block trace would usually by unit sized and split by the block size, so there would be a chance for overlapping requests. An object style with weights would require the same offset was requested, which for disk traces is a bit low. I wonder if a block size should be used rather than a weight? The readme is vague but it seems odd to me when thinking through this.

Reply:
Oh I was remembering the Umass reader which has 512 butte blocks
https://github.com/ben-manes/caffeine/blob/master/simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/parser/umass/storage/StorageTraceReader.java

Reply:
It looks like modern disks increased to 4KiB block sizes a while back, so that was likely as this is a newer trace. If you don’t mind, I’ll probably switch to that and only keep a latency aware version. I think a weighted trace likely wouldn’t be a very realistic workload given how the disk caches actually work.

Reply:
No problem, sounds very reasonable...
Do you want me to do it and then push the changes, or you will change it post merge?

Reply:
I can do it post merge

Reply:
and BTW I went over more than 50% of the offsets in all of the trace files and still no offset that is too big to fit in a long value (i.e. all of them so far are less than 64 bits)

Reply:
Great, then let’s keep your truncation as seems fine and simple

Reply:
I'm inclined to agree, since I assume that the traces are from a 64-bit system, thus all of the offsets should fit in 64 bits.

Reply:
If after the analisys is over I find some problem with this, I could do a PR with a fix using a hash (I don't think it will be needed).
So on my side we can merge.

Reply:
Thanks. It’s 1am PST so I’ll squash, merge, and do a pass of small changes tomorrow for both of your PRs. Thanks for the improvements.